internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -145,25 +145,33 @@ func resourceKustoCluster() *pluginsdk.Resource {
 			"virtual_network_configuration": {
 				Type:     pluginsdk.TypeList,
 				Optional: true,
-				ForceNew: true,
 				MaxItems: 1,
 				Elem: &pluginsdk.Resource{
 					Schema: map[string]*pluginsdk.Schema{
 						"subnet_id": {
 							Type:         pluginsdk.TypeString,
 							Required:     true,
+							ForceNew:     true,
 							ValidateFunc: commonids.ValidateSubnetID,
 						},
 						"engine_public_ip_id": {
 							Type:         pluginsdk.TypeString,
 							Required:     true,
+							ForceNew:     true,
 							ValidateFunc: networkValidate.PublicIpAddressID,
 						},
 						"data_management_public_ip_id": {
 							Type:         pluginsdk.TypeString,
 							Required:     true,
+							ForceNew:     true,
 							ValidateFunc: networkValidate.PublicIpAddressID,
 						},
+						"state": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							Default:      clusters.VnetStateEnabled,
+							ValidateFunc: validation.StringInSlice(clusters.PossibleValuesForVnetState(), false),
+						},
```

Comment:
Instead of changing the block would it make more sense to expose the `state` as a top level field called something like `virtual_network_enabled`?